dotenv.go

```diff
@@ -0,0 +1,218 @@
+package dotenv
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+const (
+	posKey int = iota
+	posVal
+)
+
+// ErrInvalidln indicates an invalid line
+var ErrInvalidln = errors.New("invalid line")
```

Comment:
I think your errors are self documenting for the most part.  I would group them and provide one comment above the group (provide individual comments if absolutely necessary.  That's just my two cents though.  It's fine as is.
```
var (
    ErrInvalidln = errors.New("invalid line")
    ...
)
```

Reply:
I think the tools complained without individual comments, but I see that goes away if you comment a block of declarations.